src/transformers/onnx/features.py

```diff
@@ -277,45 +277,68 @@ def feature_to_task(feature: str) -> str:
         return feature.replace("-with-past", "")
 
     @staticmethod
-    def get_model_class_for_feature(feature: str) -> Type:
+    def _validate_framework_choice(framework: str):
+        """
+        Validates if the framework requested for the export is both correct and available, otherwise throws an
+        exception.
+        """
+        if framework not in ["pt", "tf"]:
+            raise ValueError(
+                f"Only two frameworks are supported for ONNX export: pt or tf, but {framework} was provided."
+            )
+        elif framework == "pt" and not is_torch_available():
+            raise RuntimeError("Cannot export model to ONNX using PyTorch because no PyTorch package was found.")
+        elif framework == "tf" and not is_tf_available():
+            raise RuntimeError("Cannot export model to ONNX using TensorFlow because no TensorFlow package was found.")
+
+    @staticmethod
+    def get_model_class_for_feature(feature: str, framework: str = "pt") -> Type:
         """
         Attempt to retrieve an AutoModel class from a feature name.
 
         Args:
             feature: The feature required.
+            framework: The framework to use for the export.
 
         Returns:
             The AutoModel class corresponding to the feature.
         """
         task = FeaturesManager.feature_to_task(feature)
-        if task not in FeaturesManager._TASKS_TO_AUTOMODELS:
+        FeaturesManager._validate_framework_choice(framework)
+        if framework == "pt":
+            task_to_automodel = FeaturesManager._TASKS_TO_AUTOMODELS
+        else:
+            task_to_automodel = FeaturesManager._TASKS_TO_TF_AUTOMODELS
+        if task not in task_to_automodel:
             raise KeyError(
                 f"Unknown task: {feature}. "
                 f"Possible values are {list(FeaturesManager._TASKS_TO_AUTOMODELS.values())}"
```

Comment:
What happens here if only `tensorflow` is installed? 

If I understand correctly, since `_TASKS_TO_AUTOMODELS` is an empty `dict` by default [here](https://github.com/huggingface/transformers/pull/16018/files#diff-f4bad33d844c5d91b09dd0af27395ade640e907920b357daf125dd19458bdd60R83), this error won't show the possible values for a pure `tensorflow` env because we should be accessing `_TASKS_TO_TF_AUTOMODELS` in that case

Reply:
So I added a method called `_validate_framework_choice` that should take care of that.

Reply:
Nice!